src/Integration/Language.cs

```diff
@@ -127,13 +115,8 @@ public static Language ForProject(EnvDTE.Project dteProject)
                 throw new ArgumentNullException(nameof(dteProject));
             }
 
-            Guid projectKind;
-            if (!Guid.TryParse(dteProject.Kind, out projectKind))
-            {
-                return Unknown;
-            }
-
-            return KnownLanguages.FirstOrDefault(x => x.ProjectType == projectKind) ?? Unknown;
+            // TODO: remove this method - change the callers to call the following method directly
+            return ProjectToLanguageMapper.GetLanguageForProject(dteProject);
```

Comment:
The method _Language.ForProject_ is called from many places in the code and tests. I haven't updated all of the callers yet to make these changes easier to review. I'll remove this method in a separate commit.